### PR TITLE
nixos: mpd: fixes and socket activation

### PR DESCRIPTION
**Pull request by ghost**
setting group to audio in mpd config fixes alsa permissions issues
use real-time priority configuration from upstream
move client socket configuration to systemd, use socket activation

My first concern was fixing ALSA permissions problems, but then I looked at upstream's systemd files.
Socket activation is cool and boots a system a little faster, but I can revert that if people don't like it.

@dmalikov you touched the network config last, are you ok with socket activation?

### CODE REVIEW

**Comment by dmalikov** [comment]
:+1:

**Comment by lethalman** [comment]
cc @shlevy

**Comment by shlevy** [comment]
:+1:

File: nixos/modules/services/audio/mpd.nix [hunk 1]
```diff
@@ -31,13 +26,18 @@ in {
...
         description = ''
           Whether to enable MPD, the music player daemon.
+          MPD will be socket activated, started on demand when the first client connects.
+          Set <literal>systemd.services.mpd.wantedBy</literal> to <literal>[ "multi-user.target" ]</literal>
+          to start explicitly at boot.
         '';
       };
 
       musicDirectory = mkOption {
+        type = types.str;
```
Comment (offlinehacker) [inline nixos/modules/services/audio/mpd.nix]: use types.path here

File: nixos/modules/services/audio/mpd.nix [hunk 2]
```diff
@@ -55,6 +56,7 @@ in {
       };
 
       dataDir = mkOption {
+        type = types.str;
```
Comment (offlinehacker) [inline nixos/modules/services/audio/mpd.nix]: types.path

**Comment by ehmry** [comment]
done

File: nixos/modules/services/audio/mpd.nix [hunk 3]
```diff
@@ -62,26 +64,15 @@ in {
...
-          description = ''
-            This setting is the TCP port that is desired for the daemon to get assigned
-            to.
-          '';
-        };
-
+      listenStreams = mkOption {
+        type = types.listOf types.str;
+        default = [ "/run/mpd/socket" "6600" ];
+        example = [ "/run/mpd/socket" "19.168.1.32:6600"];
```
Comment (matthiasbeyer) [inline nixos/modules/services/audio/mpd.nix]: I guess you meant `192.168.1.32:6600` here?

**Comment by matthiasbeyer** [comment]
@lethalman If you do not care about my latest annotation this seems to be ready.

File: nixos/modules/services/audio/mpd.nix (hunk 3 above)
Comment (ehmry) [inline nixos/modules/services/audio/mpd.nix]: No, 6600 just means bind to all interfaces, just like the default mpd configuration.

Comment (ehmry) [inline nixos/modules/services/audio/mpd.nix]: I updated the description there to clarify that a little.

**Comment by ehmry** [comment]
New version came out January 14, so I bumped, and I added CIFS support that is off by default. All the other options are on by default but I left CIFS off by default because I figure samba is pretty heavy.

**Comment by ehmry** [comment]
Ok, client support allows a mpd instance to reuse the database of another mpd intance.

**Comment by bennofs** [comment]
What are the advantages of using `user mpd` in the configuration vs using systemd to to start mpd directly under the `mpd` user?

**Comment by ehmry** [comment]
@bennofs I'm not sure if has any effect, systemd starts mpd as the mpd user.

**Pull request closed without merging**